src/main/java/com/fasterxml/jackson/core/io/BigDecimalParser.java

```diff
@@ -33,18 +33,11 @@ public static BigDecimal parse(String valueStr) {
     }
 
     public static BigDecimal parse(char[] chars, int off, int len) {
-        if (off > 0 || len != chars.length) {
-            chars = Arrays.copyOfRange(chars, off, off+len);
-        }
-        return parse(chars);
-    }
-
-    public static BigDecimal parse(char[] chars) {
-        final int len = chars.length;
         try {
             if (len < 500) {
-                return new BigDecimal(chars);
+                return new BigDecimal(chars, off, len);
             }
+            chars = Arrays.copyOfRange(chars, off, off+len);
```

Comment:
Would it make more sense to avoid this Arrays.copyOfRange and change the BigDecimalParser constructor to take (chars, off, len). It looks like the parseBigDecimal can be readily changed to support an offset and explicit len.

Reply:
I can make that, thanks!

Reply:
That could make sense if easy enough to do.

Reply:
https://github.com/FasterXML/jackson-core/pull/801